src/BlockchainReader/Controller.ts

```diff
@@ -64,19 +64,37 @@ export class Controller {
     const matchingAnchors = filter(anchorPrefixAndVersionMatchConfiguration, anchors)
     const unmatchingAnchors = reject(anchorPrefixAndVersionMatchConfiguration, anchors)
 
+    const previousBlockHash = block.previousblockhash
+
+    const previousHashByHeight = await this.dao.findHashByHeight(blockHeight - 1)
+
     logger.trace(
       {
-        blockHeight,
         blockHash,
+        blockHeight,
+        previousBlockHash,
+        previousHashByHeight,
         matchingAnchors,
         unmatchingAnchors,
       },
       'Block retrieved and scanned successfully'
     )
 
-    await this.dao.upsertEntryByHeight({
-      blockHeight,
+    if (previousHashByHeight && previousHashByHeight !== previousBlockHash)
```

Comment:
We add this `previousHashByHeight !== undefined` check (implicitly) to avoid a false warning when running the Po.et node for the first time.

Reply:
Is this because previousBlockHash could also be undefined during startup?